video/out/opengl/user_shaders.h

```diff
@@ -68,10 +73,32 @@ struct gl_user_shader {
     int components;
 };
 
-// Parse the next shader pass from `body`. The `struct bstr` is modified by the
+struct gl_user_shader_tex {
+    struct bstr name;
+    int w, h, d;
+    int components;
+    int bytes;
+    int mpgl_type;
+    GLenum gl_target;
+    GLenum gl_filter;
+    GLenum gl_border;
+    void *texdata;
+    // for video.c
+    GLenum gl_tex;
+};
+
+struct gl_user_shader {
+    enum user_shader_type type;
+    union {
+        struct gl_user_shader_hook hook;
+        struct gl_user_shader_tex tex;
+    } v;
```

Comment:
What do you suggest instead? Tagged unions are the only way to implement alternatives in C, unless you want me to eliminate the intermediate data type by having `opengl/video.c` pass callbacks to the `ser_shader.c` code instead? That would be totally doable

(Basically turning `data UserShader = UserTexture {texdata} | UserHook {hookdata}` into `type UserShader = forall r. (texdata -> r) -> (hookdata -> r) -> r`)